### ISSUE TITLE
All the things 3D

### ISSUE THREAD

**Issue by captkirk88** [craftworkgames/MonoGame.Extended#530]
Just a feature list request:
* Sprite3D - 2D texture in 3D space or even a Model.
* Batcher3D - same as Batcher2D but can draw models or Sprite3Ds/Quads.
* PerspectiveCamera (3D) - or Camera3D
* Gui components 3D? - or projected 2D Gui on a quad that allows input

**Comment by captkirk88** [comment]
I wish I would've read your comment an hour ago.  I already made my own class Poly3D that handles many vertices, still not complete though the way I want it but it does render a texture in 3D space.  I also created a PerspectiveCamera (hardly controllable with input right now) and a OrthographicCamera.  Learned a lot in the last 7 hours.

For Sprite3D it was more of a idea I had to create sprites similar to Paper Mario.

For Batch3D you got the jist of it.  Essentially just a way to draw primitive shapes and textures in 3D space without having to use a construct like Sprite.

**Comment by craftworkgames** [comment]
I agree, it's about time we started thinking about having better support for 3D games with Extended.

To what degree, I'm not sure. I think we need to start simple and carefully consider what features we introduce into the library. It could be very easy to get carried away with it.

The cameras are a good place to start. I've already renamed the camera to `OrthographicCamera` in the `devleop` branch and created an empty base class. 

As with all new features we need to start by creating a demo. It would be interesting to see how far we could get creating a little 3D FPS demo. Even just thinking about how a very crude implementation might work I can think of a bunch of things we could add to the library.
